ui/oopsyraidsy/data/00-misc/general.ts

```diff
@@ -155,16 +158,30 @@ const triggerSet: OopsyTriggerSet<Data> = {
             },
           };
         }
-        data.raiseTracker[matches.targetId] ??= data.ShortName(matches.source);
+        data.raiseCastTracker[matches.targetId] ??= data.ShortName(matches.source);
       },
     },
     {
-      id: 'General Raise Cleanup',
+      id: 'General Raise Gain',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: '94' }),
+      run: (data, matches) => {
+        ((data.raiseGainTracker ??= {})[matches.targetId] ??= []).push(matches.timestamp);
+      },
+    },
+    {
+      id: 'General Raise Removal',
       type: 'LosesEffect',
       netRegex: NetRegexes.losesEffect({ effectId: '94' }),
+      delaySeconds: 0.5,
```

Comment:
Does this have issues if two people try to raise at the same time, since this clears everything from the gain tracker, so the second raise will not be found?

Wondering if it would make more sense to instead of "all of the last gain timestamps" to have a single "last removal timestamp", and throw an error if a gain comes in after that with the same timestamp.

Reply:
i noticed that might be an issue too during my attempted rewrite, so it's working a bit differently there